CMakeLists.txt

```diff
@@ -1241,6 +1241,16 @@ else()
   set(POCL_LIBRARY_NAME "OpenCL")
 endif()
 
+# make sure the OpenCL library is found (poclcc links against it)
+if(ENABLE_ICD)
+  if(WIN32)
+    string(REGEX REPLACE "lib" "bin" OPENCL_BINDIR ${OPENCL_LIBDIR})
```

Comment:
I don't think anybody is wrong, this just seems like a missing feature. `OPENCL_LIBDIR` points to `/path/to/OpenCL/lib`, which is indeed the directory that contains the `libOpenCL.dll.a`. The `pkgconfig` file looks as follows:

```
❯ cat lib/pkgconfig/OpenCL.pc
prefix=${pcfiledir}/../..
exec_prefix=${prefix}
libdir=${exec_prefix}/lib

Name: OpenCL
Description: Khronos OpenCL ICD Loader
Requires: OpenCL-Headers
Version: 3.0
Libs: -L${libdir} -lOpenCL
```

... which also seems correct. I could add a `bindir` here, but that doesn't seem like a commonly-used thing, and `FindPkgConfig.cmake` doesn't use it anyway.

Looking at `FindPkgConfig.cmake` I guess I could clean this up a little by deriving `OPENCL_BINDIR` from `OPENCL_PREFIX`. I agree that this looks weird though, and I expected there to be a canonical way to populate `link_directories` with a `pkgconfig`-derived path to the `bin` directory, but I couldn't find anything...

Reply:
> OPENCL_LIBDIR points to /path/to/OpenCL/lib, which is indeed the directory that contains the libOpenCL.dll.a. 

If that's the case, what the `OPENCL_BINDIR` is for? Are there libraries in there (and why) the PoCL tools needs?

Reply:
> what the OPENCL_BINDIR is for

It contains `OpenCL.dll`. On Windows, runtime libraries (DLLs) are shipped in `bin` folders, while compile-time libraries (`.dll.a`) are shipped in `lib`. But we link dynamically against OpenCL, as we should, so we need the `bin` directory.

Reply:
That's strange. So an OpenCL vendor packs pkg-config.pc even though it does not make sense because one can't query the location of .dlls using pkg-config(?).

Reply:
That seems to be the way things are done, as far as I can tell.

 FWIW, this isn't an OpenCL vendor, but Khronos' ICD loader. And Hwloc has a similar pkgconfig file.
Or, for a an example not packaged by us (JuliaLang): `file` from msys2:

```
Administrator in 🌐 taurus in ~
❯ cat /usr/lib/pkgconfig/libmagic.pc
prefix=/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include

Name: libmagic
Description: Magic number recognition library
Version: 5.45
Libs: -L${libdir} -lmagic
Libs.private: -lzstd -llzma -lbz2 -lz
Cflags: -I${includedir}

Administrator in 🌐 taurus in ~
❯ pacman -Qo /usr/lib/pkgconfig/libmagic.pc
/usr/lib/pkgconfig/libmagic.pc is owned by file 5.45-3

Administrator in 🌐 taurus in ~
❯ pacman -Ql file | grep dll
file /usr/bin/msys-magic-1.dll
file /usr/lib/libmagic.dll.a
```

Again, no `bindir` in the pkgconfig pointing to `/usr/bin` where the `magic` DLL resides.

Reply:
Alright, it seems that DLLs are commonly placed under bin directory.

Now, after reading up articles about dynamic linking on Windows, it seems the the link flags provided by the `lib/pkgconfig/OpenCL.pc` in the above should be enough for linking against the OpenCL DLL (indirectly using its import library) [[1](https://nullprogram.com/blog/2021/05/31/)][[2](https://www.transmissionzero.co.uk/computing/building-dlls-with-mingw/)][[3](https://web.archive.org/web/20200519042130/http://www.mingw.org/wiki/sampleDLL)].